src/ophyd_async/epics/pmac/_pmacTrajectory.py

```diff
@@ -0,0 +1,155 @@
+import time
+
+from bluesky.protocols import Flyable, Preparable
+from scanspec.specs import Frames
+
+from ophyd_async.core.async_status import AsyncStatus, WatchableAsyncStatus
+from ophyd_async.core.signal import observe_value
+from ophyd_async.core.utils import WatcherUpdate
+from ophyd_async.epics.motion import Motor
+from ophyd_async.epics.pmac import Pmac
+
+TICK_S = 0.000001
+
+
+class PmacTrajectory(Pmac, Flyable, Preparable):
+    """Device that moves a PMAC Motor record"""
+
+    def __init__(self, prefix: str, cs: int, name="") -> None:
+        # Make a dict of which motors are for which cs axis
+        self._fly_start: float
+        self.cs = cs
+        super().__init__(prefix, cs, name=name)
+
+    async def _ramp_up_velocity_pos(
```

Comment:
nit: On the motor record I refer to this as `run_up_distance `rather than `ramp_up`. I don't have a preference for which is better, but it would be nice if they're the same